hibernate-core/src/main/java/org/hibernate/type/descriptor/jdbc/XmlAsStringJdbcType.java

```diff
@@ -179,19 +189,34 @@ public <X> ValueExtractor<X> getExtractor(JavaType<X> javaType) {
 
 				@Override
 				protected X doExtract(ResultSet rs, int paramIndex, WrapperOptions options) throws SQLException {
-					return getObject( rs.getNString( paramIndex ), options );
+					if (options.getSession().getJdbcServices().getDialect().supportsNationalizedMethods()) {
+						return getObject( rs.getNString( paramIndex ), options );
+					}
+					else {
+						return getObject( rs.getString( paramIndex ), options );
+					}
 				}
 
 				@Override
 				protected X doExtract(CallableStatement statement, int index, WrapperOptions options)
 						throws SQLException {
-					return getObject( statement.getNString( index ), options );
+					if (options.getSession().getJdbcServices().getDialect().supportsNationalizedMethods()) {
```

Comment:
I think all these should just be `options.getDialect().supportsNationalizedMethods()`, right?

Reply:
I don't see the options.getDialect() method directly

Reply:
Can you take a look at how to fix this test? https://github.com/hibernate/hibernate-orm/pull/8611#issuecomment-2203412965

Reply:
`WrapperOptions.getDialect()` was added by HHH-18206

Reply:
Ah, I see it now. It was merged into the main branch. This is a pull request for the 6.5 branch. If it were also backported to 6.5, I could fix the failing test as well.

Reply:
The PR should target `main`.